apps/condo/domains/user/utils/serverSchema/validateHelpers.js

```diff
@@ -0,0 +1,122 @@
+const { isString, isEmpty } = require('lodash')
+const isArray = require('lodash/isArray')
+
+const { GQLError } = require('@open-condo/keystone/errors')
+
+const {
+    MIN_COUNT_OF_DIFFERENT_CHARACTERS_IN_PASSWORD,
+    MAX_PASSWORD_LENGTH, MIN_PASSWORD_LENGTH,
+} = require('@condo/domains/user/constants/common')
+const { GQL_ERRORS: ERRORS } = require('@condo/domains/user/constants/errors')
+
+
+const UNICODE_ChARS_REGEX = /[\w\W]/gu
+
+/**
+ * Checks for the presence of a substring in a string, case insensitive
+ *
+ * @param str {string}
+ * @param substr {string}
+ * @return {boolean}
+ */
+const hasSubstring = (str, substr) => {
+    if (!isString(str) || isEmpty(str)) return false
+    if (!isString(substr) || isEmpty(substr)) return false
+    return str.toLowerCase().includes(substr.toLowerCase())
+}
+
+/**
+ * Counts the number of unique characters in a string and compares with the minimum allowed number
+ *
+ * @param str {string}
+ * @param count {number}
+ * @return {boolean}
+ * @constructor
+ */
+const hasDifferentCharacters = (str, count = 0) => {
+    if (!isString(str) || isEmpty(str)) return false
+
+    const chars = str.match(UNICODE_ChARS_REGEX)
+    return isArray(chars) && new Set(chars).size >= count
+}
+
+/**
+ * Some characters may have length 2.
+ *
+ * This method takes these cases into account and returns the correct length value.
+ *
+ *
+ * @example
+ *  Use length:
+ *
+ *      '🤯🤯🤯'.length --> 6
+ *      'AAA'.length --> 3
+ *
+ *  Use getStringLength:
+ *
+ *      getStringLength('🤯🤯🤯') --> 3
+ *      getStringLength('AAA') --> 3
+ *
+ * @param str {string}
+ * @return {number}
+ */
+const getStringLength = (str) => {
+    if (!isString(str) || isEmpty(str)) return 0
+
+    const chars = str.match(UNICODE_ChARS_REGEX)
+    return isArray(chars) ? chars.length : 0
+}
+
+/**
+ *
+ *
+ * @param str {string}
+ * @return {boolean}
+ */
+const hasSpacesAtStartOrAtEnd = (str) => {
+    if (!isString(str) || isEmpty(str)) return false
+
+    return str.trim() !== str
+}
+
+const passwordValidations = async (context, pass, email, phone, name) => {
+    // Password must be in string format
+    if (!isString(pass)) {
+        throw new GQLError(ERRORS.WRONG_PASSWORD_FORMAT, context)
+    }
+
+    // Password must not start or end with a space
+    if (hasSpacesAtStartOrAtEnd(pass)) {
+        throw new GQLError(ERRORS.PASSWORD_CONTAINS_SPACES_AT_BEGINNING_OR_END, context)
```

Comment:
I think we don't need to check that. it adds complexity and doesn't add product value.